test/with_dummyserver/test_socketlevel.py

```diff
@@ -239,6 +239,34 @@ def socket_handler(listener):
         finally:
             timed_out.set()
 
+    def test_delayed_body_read_timeout_with_preload(self):
+        timed_out = Event()
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+            buf = b''
+            body = 'Hi'
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+            sock.send(('HTTP/1.1 200 OK\r\n'
+                       'Content-Type: text/plain\r\n'
+                       'Content-Length: %d\r\n'
+                       '\r\n' % len(body)).encode('utf-8'))
+
+            timed_out.wait()
+            sock.send(body.encode('utf-8'))
```

Comment:
We should avoid this `send` call because there's a risk it'll cause an exception to be thrown and prevent the cleanup of the thread. Just go straight to the close call. =)